lib/kinesis_client/stream/shard/producer.ex

```diff
@@ -214,14 +198,22 @@ defmodule KinesisClient.Stream.Shard.Producer do
     meta = telemetry_meta(state)
 
     :telemetry.span([:kinesis_client, :shard_producer, :update_checkpoint], meta, fn ->
-      :ok =
-        AppState.update_checkpoint(
-          state.app_name,
-          state.shard_id,
-          state.lease_owner,
-          checkpoint,
-          state.app_state_opts
-        )
+      case AppState.update_checkpoint(
+             state.app_name,
+             state.shard_id,
+             state.lease_owner,
+             checkpoint,
+             state.app_state_opts
+           ) do
+        :ok ->
+          :ok
+
+        {:error, :lease_owner_match} ->
+          raise "Checkpoint failed for #{state.app_name}-#{state.shard_id}: this consumer is not the lease owner"
+
+        unknown ->
+          raise "Checkpoint failed for #{state.app_name}-#{state.shard_id}: #{inspect(unknown)}"
```

Comment:
Same suggestion: I think custom exceptions will aide us in debugging Sentry reports.